docs/index.md

```diff
@@ -6,69 +6,87 @@ This page is the top level entry point into the Substrate documentation.
+> eventually, we aim for this not to be the case).
 
 Substrate is a blockchain development framework with a completely generic State Transition Function
-([STF](knowledgebase/getting-started/glossary#stf-state-transition-function)) and modular components for consensus, networking,
-and configuration.
-
-Despite being "completely generic", it comes with both standards and conventions - particularly with
-the Substrate runtime module library (a.k.a [FRAME](knowledgebase/runtime/frame.md)) - regarding the underlying
-data-structures that power the STF, thereby making rapid blockchain development a reality.
+([STF](knowledgebase/getting-started/glossary#stf-state-transition-function)) and modular components for consensus,
+networking, and configuration.
+
+Despite being "completely generic", it comes with both standards and conventions - particularly with the Substrate
+runtime module library (a.k.a [FRAME](knowledgebase/runtime/frame.md)) - regarding the underlying data-structures that
+power the STF, thereby making rapid blockchain development a reality.
+
+## Architecture
+
+![Substrate Client Architecture](assets/substrate-arch.png)
+
+The Substrate client is an application that runs a Substrate-based blockchain node - it consists of several components
+that include, but are not limited to, the following:
+
+- **Storage** is used to persist the evolving state of the decentralized system represented by a blockchain. The
+  blockchain network allows participants to reach trustless [consensus](knowledgebase/advanced/consensus) about the
+  state of storage. Substrate ships with a simple and highly efficient
+  [key-value storage mechanism](knowledgebase/advanced/storage).
+- **Runtime** logic defines how blocks are processed, including state transition logic. In Substrate, runtime code is
+  compiled to [Wasm](knowledgebase/getting-started/glossary#webassembly-wasm) and becomes part of the blockchain's
+  storage state - this enables one of the defining features of a Substrate-based blockchain:
+  [forkless runtime upgrades](knowledgebase/advanced/executor#forkless-runtime-upgrades). Substrate clients may also
+  include a "native runtime" that is compiled for the same platform as the client itself (as opposed to Wasm).
+- **Peer-to-peer network** capabilities allow the client to communicate with other network participants. Substrate uses
+  [the `libp2p` network stack](https://libp2p.io/).
+- **Consensus** engines provide logic that allows network participants to agree on the state of the blockchain.
+  Substrate makes it possible to supply custom consensus engines and also ships with several consensus mechanisms that
+  have been built on top of [Web3 Foundation research](https://w3f-research.readthedocs.io/en/latest/index.html).
+- **Remote procedure calls** allow blockchain users to interact with the network. Substrate provides HTTP and WebSocket
```

Comment:
No one actually uses the term "Remote procedure calls" right? Just say RPC, which seems pretty colloquial.

Reply:
I might suggest doing both- say RPC (Remote Procedure Calls) just to define it on its first instance for newbies